src/nxt_main_process.c

```diff
@@ -1091,131 +1092,6 @@ nxt_main_port_socket_handler(nxt_task_t *task, nxt_port_recv_msg_t *msg)
 }
 
 
-static nxt_int_t
-nxt_main_listening_socket(nxt_sockaddr_t *sa, nxt_listening_socket_t *ls)
-{
-    nxt_err_t         err;
-    nxt_socket_t      s;
-
-    const socklen_t   length = sizeof(int);
-    static const int  enable = 1;
-
-    s = socket(sa->u.sockaddr.sa_family, sa->type, 0);
-
-    if (nxt_slow_path(s == -1)) {
-        err = nxt_errno;
-
-#if (NXT_INET6)
-
-        if (err == EAFNOSUPPORT && sa->u.sockaddr.sa_family == AF_INET6) {
-            ls->error = NXT_SOCKET_ERROR_NOINET6;
-        }
-
-#endif
-
-        ls->end = nxt_sprintf(ls->start, ls->end,
```

Comment:
The logic is required but dropped. `start` and `end` is used.
```
ls->end = nxt_sprintf(ls->start, ls->end, "...");
```

Reply:
Hello @hongzhidao,

As far as I can see, this logic is not actually used during compilation. Please correct me if I'm wrong.
https://github.com/nginx/unit/blob/8138d15f76b1017ba61273f69416ffbe028b0d67/src/nxt_router.c#L2659

I can implement the logic you mentioned if you think it's necessary in all other related places in the code.

Reply:
You can test with an existing listening sock file.
```
    ret = nxt_listen_socket_create(task, task->thread->runtime->mem_pool, &ls);

    if (ret == NXT_OK) {
        ...

    } else {
        size = ls.end - ls.start;

        nxt_alert(task, "%*s", size, ls.start);  // ls.start is used if it failed.
```

Reply:
You're right! Thank you for bringing this to my attention. I'm going to fix it.